SimFastTiming/FastTimingCommon/src/BTLElectronicsSim.cc

```diff
@@ -16,30 +34,95 @@ BTLElectronicsSim::BTLElectronicsSim(const edm::ParameterSet& pset) :
 
 
 void BTLElectronicsSim::run(const mtd::MTDSimHitDataAccumulator& input,
-			    BTLDigiCollection& output) const {
-  
-  MTDSimHitData chargeColl,toa;
+			    BTLDigiCollection& output,
+			    CLHEP::HepRandomEngine *hre) const {
   
+  MTDSimHitData chargeColl, toa1, toa2;
+
   for(MTDSimHitDataAccumulator::const_iterator it=input.begin();
       it!=input.end();
       it++) {
     
-    chargeColl.fill(0.f); 
-    toa.fill(0.f);
+    chargeColl.fill(0.f);
+    toa1.fill(0.f);
+    toa2.fill(0.f);
     for(size_t i=0; i<it->second.hit_info[0].size(); i++) {
-      //time of arrival
-      float finalToA = (it->second).hit_info[1][i];
-      while(finalToA < 0.f)  finalToA+=25.f;
-      while(finalToA > 25.f) finalToA-=25.f;
-      toa[i]=finalToA;
+
+      // --- Fluctuate the total number of photo-electrons
+      float Npe = CLHEP::RandPoissonQ::shoot(hre, (it->second).hit_info[0][i]);
+      if ( Npe < EnergyThreshold_ ) continue;
+
+
+      // --- Get the time of arrival and add a channel time offset
+      float finalToA1 = (it->second).hit_info[1][i] + ChannelTimeOffset_;
+      float finalToA2 = (it->second).hit_info[1][i] + ChannelTimeOffset_;
```

Comment:
`finalToA2 = finalToA1`?

Reply:
Yes. They get different values later, when the time-walk is simulated and added.
I've rearranged the code to avoid duplicate operations.